src/transaction.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Account,
+  FeeBumpTransaction,
+  Operation,
+  Transaction,
+  TransactionBuilder,
+  xdr,
+} from "stellar-base";
+
+// TODO: Transaction is immutable, so we need to re-build it here. :(
+export function prepareTransaction(
```

Comment:
a unit test demonstrating input/output would be nice if feasible, the visibility from that could also double as a developer reference for latest doc on `SimulateTransactionResponse.results`.